src/Collision.cc

```diff
@@ -200,6 +218,73 @@ sdf::SemanticPose Collision::SemanticPose() const
       this->dataPtr->poseRelativeToGraph);
 }
 
+/////////////////////////////////////////////////
+Errors Collision::CalculateInertial(gz::math::Inertiald &_inertial,
+                                    const ParserConfig &_config)
+{
+  Errors errors;
+
+  // Set the density value for the collision material
+  if (this->dataPtr->sdf->HasElement("density"))
+  {
+    this->dataPtr->density = this->dataPtr->sdf->Get<double>("density");
+  }
```

Comment:
This should be done in `Load` since `CalculateInertial` might be called on an object created programmatically, i.e., `dataPtr->sdf` is nullptr.

```
sdf::Collision foo;
foo.SetDensity(1.0);
foo.CalculateInerital(...);
```

The error/warning message would still need to be here because it should only be emitted if we're doing automatic inertias. You can set a flag in `Load` to capture whether `density` was set from SDF or not.

Reply:
Removed `dataPtr->sdf` based check from the function in [a658c25](https://github.com/gazebosim/sdformat/pull/1299/commits/a658c25f76cd67a881c2a6f6dcb1017a753ceba2). Also added a separate flag variable to check if the density value provided in the sdf.